src/toil/test/__init__.py

```diff
@@ -585,27 +586,7 @@ def make_tests(generalMethod, targetClass=None, **kwargs):
     >>> hasattr(f, 'test_has__num_one__letter_a')  # should be false because Foo has no test methods
     False
 
-    >>> make_tests(Foo.has, num={'one':1, 'two':2}, letter={'a':'a', 'b':'b'})
```

Comment:
This is the best illustration of the problem I've found: https://stackoverflow.com/questions/3589311/get-defining-class-of-unbound-method-object-in-python-3

Basically, python 3 gets rid of the concept of "unbound method" and instead just has "functions".

The `make_tests` method, when `targetClass` is None relies on this behavior. The only place that is used is in these tests, so it seems safe to delete? Perhaps we should make `targetClass` non-optional to make the change more clear.